Dockerfile

```diff
@@ -0,0 +1,70 @@
+# ===============================================
+FROM helsinkitest/node:12-slim as appbase
+# ===============================================
+# Offical image has npm log verbosity as info. More info - https://github.com/nodejs/docker-node#verbosity
+ENV NPM_CONFIG_LOGLEVEL warn
+
+# set our node environment, either development or production
+# defaults to production, compose overrides this to development on build and run
+ARG NODE_ENV=production
+ENV NODE_ENV $NODE_ENV
+
+# Global npm deps in a non-root user directory
+ENV NPM_CONFIG_PREFIX=/app/.npm-global
+ENV PATH=$PATH:/app/.npm-global/bin
+
+# Yarn
+ENV YARN_VERSION 1.19.1
+RUN yarn policies set-version $YARN_VERSION
+
+# Use non-root user
+USER appuser
+
+# Copy package.json and package-lock.json/yarn.lock files
+COPY package*.json *yarn* ./
+
+# Install npm depepndencies
+ENV PATH /app/node_modules/.bin:$PATH
+# RUN yarn && yarn cache clean --force
+
+USER root
+
+RUN apt-install.sh build-essential
+
+USER appuser
+RUN yarn && yarn cache clean --force
+
+USER root
+RUN apt-cleanup.sh build-essential
```

Comment:
The cleanup step here will not result in much. It will remove things from the final container, but it will not make the image any smaller due to previous steps already being cached.

What would make the image smaller would be something in the lines of:
```
USER root
RUN apt-install.sh build-essential && \
         su - appuser -c "yarn && yarn cache clean --force" && \
         apt-cleanup.sh build-essential
USER appuser
```
I do however agree here that this is not the nicest thing in the word read or write.

Another way could be to install everything as root, and then chown it, that
would make things a bit more readable maybe:
```
USER root
RUN apt-install.sh build-essential && \
         yarn && yarn cache clean --force && \
         apt-cleanup.sh build-essential && \
         sudo chown -R appuser:appuser ./node_modules
USER appuser
```
This has the caveat that if things are _not_ installed to `node_modules` for any reason, then
this will not be enough and will not cover that case.

Reply:
Alright, I went with the first option :).